kraken-build/src/kraken/std/python/tasks/ruff_task.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from dataclasses import dataclass
+from pathlib import Path
+
+from kraken.common import Supplier
+from kraken.core import Project, Property
+from kraken.std.python.tasks.pex_build_task import pex_build
+
+from .base_task import EnvironmentAwareDispatchTask
+
+
+class RuffTask(EnvironmentAwareDispatchTask):
+    """A task to run `ruff` in either format, fix, or check mode."""
+
+    description = "Lint Python source files with Ruff."
+    python_dependencies = ["ruff"]
+    ruff_task: Property[str] = Property.default("")
+
+    ruff_bin: Property[str] = Property.default("ruff")
+    config_file: Property[Path]
+    additional_args: Property[Sequence[str]] = Property.default_factory(list)
+
+    def get_execute_command(self) -> list[str]:
+        command = [
+            self.ruff_bin.get(),
+            self.ruff_task.get(),
+            str(self.settings.source_directory),
+        ] + self.settings.get_tests_directory_as_args()
+        command += [str(directory) for directory in self.settings.lint_enforced_directories]
+        if self.config_file.is_filled():
+            command += ["--config", str(self.config_file.get().absolute())]
+        command += self.additional_args.get()
+        return command
+
+
+@dataclass
+class RuffTasks:
+    check: RuffTask
+    fix: RuffTask
+    format: RuffTask
+
+
+def ruff(
+    *,
+    name: str = "python.ruff",
+    project: Project | None = None,
+    config_file: Path | Supplier[Path] | None = None,
+    additional_args: Sequence[str] | Supplier[Sequence[str]] = (),
+    version_spec: str | None = None,
+    additional_requirements: Sequence[str] = (),
+) -> RuffTasks:
+    """Creates three tasks for formatting and linting your Python project with Ruff.
+
+    :param version_spec: If specified, the Ruff tool will be installed as a PEX and does not need to be installed
+        into the Python project's virtual env.
```

Comment:
Not all parameters are documented

Reply:
Initially kept same level of documentation there is in other formatter / linter. 
Added documentation for each params